app/Controllers/Http/UsersController.ts

```diff
@@ -0,0 +1,76 @@
+import type { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'
+import Profile from 'App/Models/Profile'
+import User from 'App/Models/User'
+import Hash from '@ioc:Adonis/Core/Hash'
+import CreateUser from 'App/Validators/CreateUserValidator'
+
+
+export default class UsersController {
+    public async getAllUsers() {
+        return User.all()
+    }
+
+    public async deleteUser({ request, auth, response }: HttpContextContract) {
+        try {
+            const { mobile_number } = request.all();
+            const user = auth.use('api').user!
+
+            const profile = await Profile.findByOrFail('user_id', user.id)
+            if (profile.mobileNumber === mobile_number) {
+                await user.delete()
+                return {
+                    message: 'Your profile has been Deleted'
+                }
+            } else {
+                response.status(403)
+                return {
+                    error: 'Please enter valid number'
```

Comment:
This is not needed and as it's not mentioned in the task.

Reply:
Done